bug: `pytest -m core` fails on main due to missing `black`package

### What happened?

I followed the [contribution guideline](https://ibis-project.org/contribute/01_environment) to set up a development environment.  When running `pytest -m core` multiple tests (`ibis/expr/tests/test_sql.py::test_parse_sql_*`) fail because `black` isn't installed. After installing `black` all tests pass.

### What version of ibis are you using?

main

### What backend(s) are you using, if any?

- 

### Relevant log output

```sh
FAILED ibis/expr/tests/test_sql.py::test_parse_sql_aggregation_with_multiple_joins - ImportError: The 'format' option requires the 'black' package to be installed
FAILED ibis/expr/tests/test_sql.py::test_parse_sql_basic_join[inner] - ImportError: The 'format' option requires the 'black' package to be installed
FAILED ibis/expr/tests/test_sql.py::test_parse_sql_basic_aggregation_with_join - ImportError: The 'format' option requires the 'black' package to be installed
FAILED ibis/expr/tests/test_sql.py::test_parse_sql_basic_join[right] - ImportError: The 'format' option requires the 'black' package to be installed
FAILED ibis/expr/tests/test_sql.py::test_parse_sql_simple_reduction - ImportError: The 'format' option requires the 'black' package to be installed
FAILED ibis/expr/tests/test_sql.py::test_parse_sql_join_with_filter - ImportError: The 'format' option requires the 'black' package to be installed
FAILED ibis/expr/tests/test_sql.py::test_parse_sql_multiple_joins - ImportError: The 'format' option requires the 'black' package to be installed
FAILED ibis/expr/tests/test_sql.py::test_parse_sql_basic_projection - ImportError: The 'format' option requires the 'black' package to be installed
FAILED ibis/expr/tests/test_sql.py::test_parse_sql_in_clause - ImportError: The 'format' option requires the 'black' package to be installed
FAILED ibis/expr/tests/test_sql.py::test_parse_sql_scalar_subquery - ImportError: The 'format' option requires the 'black' package to be installed
FAILED ibis/expr/tests/test_sql.py::test_parse_sql_table_alias - ImportError: The 'format' option requires the 'black' package to be installed
FAILED ibis/expr/tests/test_sql.py::test_parse_sql_basic_aggregation - ImportError: The 'format' option requires the 'black' package to be installed
FAILED ibis/expr/tests/test_sql.py::test_parse_sql_basic_join[left] - ImportError: The 'format' option requires the 'black' package to be installed
FAILED ibis/expr/tests/test_sql.py::test_parse_sql_simple_select_count - ImportError: The 'format' option requires the 'black' package to be installed
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/conda/environment-arm64-flink.yml
+++ b/conda/environment-arm64-flink.yml
@@ -6,6 +6,7 @@ dependencies:
   - python =3.10
   - atpublic >=2.3
   - bidict >=0.22.1
+  - black >=22.1.0,<25
   - clickhouse-connect >=0.5.23
   - dask >=2022.9.1
   - datafusion >=0.6

--- a/conda/environment-arm64.yml
+++ b/conda/environment-arm64.yml
@@ -6,6 +6,7 @@ dependencies:
   - python >=3.10
   - atpublic >=2.3
   - bidict >=0.22.1
+  - black >=22.1.0,<25
   - clickhouse-connect >=0.5.23
   - dask >=2022.9.1
   - datafusion >=0.6

--- a/conda/environment.yml
+++ b/conda/environment.yml
@@ -6,6 +6,7 @@ dependencies:
   - apache-flink
   - atpublic >=2.3
   - bidict >=0.22.1
+  - black >=22.1.0,<25
   - clickhouse-connect >=0.5.23
   - dask >=2022.9.1
   - datafusion >=0.6

--- a/docs/contribute/01_environment.qmd
+++ b/docs/contribute/01_environment.qmd
@@ -97,7 +97,7 @@ for manager, params in managers.items():
     )
     print()
     print("    ```sh")
-    print("    pip install -e '.[all]'")
+    print("    pip install -e .")
     print("    ```")
     print()
 ```
@@ -200,7 +200,7 @@ For a better development experience see the `conda/mamba` or `nix` setup instruc
 1. Install ibis in development mode
 
    ```sh
-   pip install -e '.[all]'
+   pip install -e .
    ```
 
 :::

